### DISCUSSION TITLE
Automate SO setup

### DISCUSSION THREAD

**Discussion by jLemmings** [Security-Onion-Solutions/securityonion#2747] [General] [answered]
Hello

I need to automate the SO 2.3 setup with Ansible. With SO 1 I was able to achieve this by passing a configuration file when installing. With SO 2.3 I have not found any way to do this, has this feature been removed? If so are there any alternatives you can reccomend?

Thanks

**Comment by TOoSmOotH** [comment, marked as answer]
You can look at setup/automation to automate answers but salt is still required. You would source your answer file while launching so-setup.

**Reply by jLemmings** [reply]
How do I pass the source file with my answers to so-setup? I have not found the current way to do this...

Edit: Found the way to pass the answer file: `so-setup `accepts a 2nd paremeter which should be the name of the answer file located in `securityonion/setup/automation`.

**Reply by threathunternotebook** [reply]
I haven't had much luck with this either.  I configure the automation/distributed-airgap-manager file.  Then use:
sudo ./so-setup iso distributed-airgap-manager
Looking at the the sosetup.log, it hangs at "Testing if setup is running on a cloud instance."
We are deploying SO 16.04 distributed via Ansible also on backend Vspheres that don't have internet connectivity.  I had no issues, but 2.3 is just not working out.  No documentation on trying to do an automated install either.

**Reply by jLemmings** [reply]
@threathunternotebook Try it like this: ./so-setup network distributed-airgap-manager

That is how it worked for me.

**Reply by threathunternotebook** [reply]
@jLemmings That worked for me.  The install is running.  Thanks.

**Reply by Mapmark** [reply]
Don't mean to bring back an answered discussion...

@threathunternotebook & @jLemmings . When you pass that answer file to so-setup. Is the GUI still prompted for you, or does it just run the install? I'm my scenario, I'm still prompted for the installation for the GUI even after passing. ./so-setup network <answerfile>
